packages/nextjs/src/config/webpack.ts

```diff
@@ -695,7 +695,7 @@ export function getWebpackPluginOptions(
   const defaultPluginOptions = dropUndefinedKeys({
     include: isServer ? serverInclude : clientInclude,
     ignore:
-      isServer || !userSentryOptions.widenClientFileUpload
+      isServer || userSentryOptions.widenClientFileUpload
```

Comment:
In the current implementation, the default options are set as follows:

- When `isServer == false && widenClientFileUpload === false`
  - `includes`: `` [ { paths: [`${distDirAbsPath}/static/chunks/pages`], urlPrefix: `${urlPrefix}/static/chunks/pages` }, { paths: [`${distDirAbsPath}/static/chunks/app`], urlPrefix: `${urlPrefix}/static/chunks/app` } ] ``
  - `ignore`: `[]`
  - Sourcemaps to be uploaded:
    - `.next/static/chunks/pages/xxx.js.map`
  - Sourcemaps to be **NOT** uploaded:
    - `.next/static/chunks/framework-*.js.map` (excluded by `includes` option)
    - `.next/static/chunks/main-*.js.map` (excluded by `includes` option)
- When `isServer == false && widenClientFileUpload === true`
  - `includes`: `` [{ paths: [`${distDirAbsPath}/static/chunks`], urlPrefix: `${urlPrefix}/static/chunks` }] ``
  - `ignore`: ``['framework-*', 'framework.*', 'main-*', 'polyfills-*', 'webpack-*'] ``
  - Sourcemaps to be uploaded:
    - `.next/static/chunks/pages/xxx.js.map`
  - Sourcemaps to be **NOT** uploaded:
    - `.next/static/chunks/framework-*.js.map` (**excluded by `ignore` option**)
    - `.next/static/chunks/main-*.js.map` (**excluded by `ignore` option**)

As you can see, there is a bug where sourcemaps such as `framework-*` are not uploaded when `widenClientFileUpload === true`. My change fixes this.

My change sets the default options as follows:

- When `isServer == false && widenClientFileUpload === false`
  - `includes`: `` [ { paths: [`${distDirAbsPath}/static/chunks/pages`], urlPrefix: `${urlPrefix}/static/chunks/pages` }, { paths: [`${distDirAbsPath}/static/chunks/app`], urlPrefix: `${urlPrefix}/static/chunks/app` } ] ``
  - `ignore`: `` [{ paths: [`${distDirAbsPath}/static/chunks`], urlPrefix: `${urlPrefix}/static/chunks` }] ``
  - Sourcemaps to be uploaded:
    - `.next/static/chunks/pages/xxx.js.map`
  - Sourcemaps to be **NOT** uploaded:
    - `.next/static/chunks/framework-*.js.map` (excluded by `includes` and `ignore` option)
    - `.next/static/chunks/main-*.js.map` (excluded by `includes` and `ignore` option)
- When `isServer == false && widenClientFileUpload === true`
  - `includes`: `` [{ paths: [`${distDirAbsPath}/static/chunks`], urlPrefix: `${urlPrefix}/static/chunks` }] ``
  - `ignore`: `[]`
  - Sourcemaps to be uploaded:
    - `.next/static/chunks/pages/xxx.js.map`
    - `.next/static/chunks/framework-*.js.map`
    - `.next/static/chunks/main-*.js.map`

The default value of `widenClientFileUpload` is `false`, but `.next/static/chunks/framework-*.js.map` is still excluded when `false`. This is not a breaking change.

Reply:
The thing you're suggesting is wrong, is intentional. There is some history to it because we ran into extremely long uploads and we didn't want to upload these files because they don't contain user code. These files only contain stuff like Next.js internal code, Webpack code and/or React code which is often useless when looking at stacktraces.

See comment: 

```
Widening the upload scope is necessarily going to lead to us uploading files we don't need to (ones which
don't include any user code). In order to lessen that where we can, exclude the internal nextjs files we know
will be there.
```

I guess we could rethink this...

Reply:
Oh, I understand what you are saying. I was wrong.

Errors can often occur in libraries due to bugs in the library or by using the library in the wrong way. Therefore, I think it is only natural to upload those sourcemaps. At least uploading those sourcemaps helps me.

But I agree with your opinion. I am going to close this PR.